app/models/manageiq/providers/hawkular/middleware_manager/alert_profile_manager.rb

```diff
@@ -2,8 +2,8 @@ module ManageIQ::Providers
   class Hawkular::MiddlewareManager::AlertProfileManager
     require 'hawkular/hawkular_client'
 
-    def initialize(ems)
-      @alerts_client = ems.connect.alerts
+    def initialize(alerts_client)
+      @alerts_client = alerts_client
```

Comment:
same, This line is just taking the injected argument as-is, 
but when MiddlewareManager instantiates this, it passes the result of calling `MiddlewareManager#alerts_client`.

(That `alerts_client` method does `with_provider_connection &:alerts` which boils down to `connect.alerts`.  You get the same `Hawkular::Alerts::AlertsClient` object.  `connect` caches a single connection.)